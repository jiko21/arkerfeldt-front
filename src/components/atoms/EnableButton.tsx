/** @jsxImportSource @emotion/react */
import { css } from '@emotion/react';
import { ButtonHTMLAttributes, FC } from 'react';
import { Color } from '@/const/color';

const buttonStyle = css`
  background-color: ${Color.ENABLE};
  border: none;
  box-shadow: 4px 4px 4px rgba(0, 0, 0, 0.25);
  color: #ffffff;
  font-size: 14px;
  font-weight: 700;
  height: 30px;
  min-width: 80px;
  &:hover {
    background-color: ${Color.ENABLE_HOVER};
  }
`;

interface IProps extends ButtonHTMLAttributes<HTMLButtonElement> {
  testId?: string;
}

const EnableButton: FC<IProps> = ({ children, testId, ...props }) => (
  <button css={buttonStyle} {...props} data-testid={testId}>
    {children}
  </button>
);

export default EnableButton;
