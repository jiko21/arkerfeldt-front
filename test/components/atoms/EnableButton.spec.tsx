import EnableButton from '@/components/atoms/EnableButton';
import { fireEvent, render, screen } from '@testing-library/react';

describe('EnableButton.tsx', () => {
  let onClickMock: jest.Mock;
  beforeEach(() => {
    onClickMock = jest.fn();
  });

  afterEach(() => {
    onClickMock.mockClear();
  });
  it('should correctly render', () => {
    const { container } = render(
      <EnableButton onClick={onClickMock}>Sample</EnableButton>,
    );

    expect(container).toMatchSnapshot();
  });

  it('should correctly call onClick', () => {
    const TEST_ID = 'TEST_ID';
    render(
      <EnableButton onClick={onClickMock} testId={TEST_ID}>
        Sample
      </EnableButton>,
    );
    fireEvent.click(screen.getByTestId(TEST_ID));
    expect(onClickMock).toBeCalled();
  });
});
