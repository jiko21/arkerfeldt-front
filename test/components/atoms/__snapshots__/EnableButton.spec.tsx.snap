// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EnableButton.tsx should correctly render 1`] = `
.emotion-0 {
  background-color: #007AAE;
  border: none;
  box-shadow: 4px 4px 4px rgba(0, 0, 0, 0.25);
  color: #ffffff;
  font-size: 14px;
  font-weight: 700;
  height: 30px;
  min-width: 80px;
}

.emotion-0:hover {
  background-color: #005b82;
}

<div>
  <button
    class="emotion-0"
  >
    Sample
  </button>
</div>
`;
